server/datastore/mysql/apple_mdm_test.go

```diff
@@ -5519,17 +5519,17 @@ func testMDMConfigAsset(t *testing.T, ds *Datastore) {
 	err := ds.InsertMDMConfigAssets(ctx, assets)
 	require.NoError(t, err)
 
-	a, err := ds.GetMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{fleet.MDMAssetCACert, fleet.MDMAssetCAKey})
+	a, err := ds.GetAllMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{fleet.MDMAssetCACert, fleet.MDMAssetCAKey})
 	require.NoError(t, err)
 	require.Equal(t, wantAssets, a)
 
 	// try to fetch an asset that doesn't exist
-	a, err = ds.GetMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{fleet.MDMAssetABMCert})
-	require.NoError(t, err)
-	require.Len(t, a, 0)
+	a, err = ds.GetAllMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{fleet.MDMAssetABMCert})
+	require.ErrorIs(t, err, ErrPartialResult)
```

Comment:
Ah yeah it does read a bit weird with that error name when it doesn't find anything, but that's not a blocker to me.  More of a naming issue than anything else, I think.

Reply:
@mna I like your idea of `sql.ErrNoRows`, should be `notFound` instead? just because we generally wrap those `sql` errors.

Reply:
Yeah, my initial concern with `notFound` was that if not special-cased in the Service layer it would result in a 404, but maybe that makes sense (if none are found, it is "not found", and generally I guess we'll special-case it!). Either approach works for me, whatever you feel makes more sense/is easier to check.

Reply:
thank you so much for the patience navigating this! I think we've got it

Reply:
Thank you for the live idea-bouncing session! :D